src/brsTypes/components/RoSGNode.ts

```diff
@@ -48,22 +50,40 @@ class Field {
     }
 
     setValue(value: BrsType) {
-        if (this.alwaysNotify || this.value !== value) {
+        let oldValue = this.value;
+        this.value = value;
+        if (this.alwaysNotify || oldValue !== value) {
             this.observers.map(this.executeCallbacks);
         }
-        this.value = value;
+    }
+
+    /**
+     * Adds an observer and removes any existing observers with the same name.
+     * In the case where a child and a parent have the same observer name, only the
+     * child's observer should be called.
+     */
+    createExclusiveObserver(interpreter: Interpreter, callable: Callable) {
```

Comment:
Should we move this logic into `addObserver` and just have one method for adding observers? I guess we don't want more than one callback with the same name in any case.